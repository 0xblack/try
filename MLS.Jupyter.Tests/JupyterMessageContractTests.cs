using System;
using System.Text;
using Assent;
using MLS.Jupyter.Protocol;
using NetMQ;
using Xunit;

namespace MLS.Jupyter.Tests
{
    public class JupyterMessageContractTests
    {
        private readonly Configuration _configuration;

        public JupyterMessageContractTests()
        {
            _configuration = new Configuration()
                .UsingExtension("json");

            _configuration = _configuration.SetInteractive(true);
        }

        [Fact]
        public void KernelInfoReply_contract_has_not_been_broken()
        {
            var socket = new TextSocket();
            var sender = new MessageSender(socket, new SignatureValidator("key", "HMACSHA256"));
            var kernelInfoReply = new KernelInfoReply
                                  {
                                      ProtocolVersion = "5.3",
                                      Implementation = ".NET",
                                      ImplementationVersion = "0.0.3",
                                      LanguageInfo = new LanguageInfo
                                                     {
                                                         Name = "C#",
                                                         Version = typeof(string).Assembly.ImageRuntimeVersion.Substring(1),
                                                         MimeType = "text/x-csharp",
                                                         FileExtension = ".cs",
                                                         PygmentsLexer = "c#"
                                                     }
                                  };
            var header = new Header
                         {
                             Username = Constants.USERNAME,
                             Session = "test session",
                             MessageId = Guid.Empty.ToString(),
                             MessageType = MessageTypeValues.KernelInfoReply,
                             Date = DateTime.MinValue.ToString("yyyy-MM-ddTHH:mm:ssZ"),
                             Version = "5.3"
                         };
            var replyMessage = new Message
                               {
                                   Header = header,
                                   Content = kernelInfoReply
                               };
            sender.Send(replyMessage);

            var encoded = socket.GetEncodedMessage();
            this.Assent(encoded, _configuration);
        }

        private class TextSocket : IOutgoingSocket
        {
            readonly StringBuilder _buffer = new StringBuilder();

            public bool TrySend(ref Msg msg, TimeSpan timeout, bool more)
            {
                var decoded = SendReceiveConstants.DefaultEncoding.GetString(msg.Data);
                _buffer.AppendLine($"data: {decoded} more: {more}");
                return true;
            }

            public string GetEncodedMessage()
            {
                return _buffer.ToString();
            }
        }
    }
}