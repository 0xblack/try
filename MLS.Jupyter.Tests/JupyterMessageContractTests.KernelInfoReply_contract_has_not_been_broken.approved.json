data: <IDS|MSG> more: True
data: 8a1a48686cad024f4221075a0540cded362854469fc629cc1361f4459af4c35e more: True
data: {"msg_id":"2d89b7f4-06ae-4c62-ada0-8562595e7599","username":"dotnet_kernel","session":"test session","date":"0001-01-01T00:00:00Z","msg_type":"kernel_info_reply","version":"5.3"} more: True
data: null more: True
data: {} more: True
data: {"protocol_version":"5.3","implementation":".NET","implementation_version":"0.0.3","language_info":{"name":"C#","version":"4.0.30319","mimetype":"text/x-csharp","file_extension":".cs","pygments_lexer":"c#","nbconvert_exporter":null,"help_links":[]},"banner":null} more: False
